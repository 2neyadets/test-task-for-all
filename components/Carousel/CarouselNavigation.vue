<template lang="pug">
  .absolute.flex.carousel__navigation
    .carousel__navigation-inner.flex.justify-center
      template(v-for="(slide, index) in amountOfSlides")
        Btn.q-mx-xs(
          @click.native="$emit('changeSlide', index)"
          icon="brightness_1"
          :size="12"
          :active="index === currentSlide"
          round
        )
</template>

<script>

import Btn from '../Reusable/Btn'
export default {
  name: 'CarouselNavigation',
  components: { Btn },
  props: {
    currentSlide: {
      type: Number,
      required: true,
    },
    amountOfSlides: {
      type: Number,
      required: true,
    },
  },
  data () {
    return {
    }
  },
  computed: {},
  watch: {},
  created () {},
  mounted () {},
  updated () {},
  beforeDestroy () {},
  methods: {},
}
</script>

<style scoped lang="stylus">
  .carousel__navigation
    left 16px
    right 16px
    bottom 16px
    overflow-x auto
    overflow-y hidden

    &-inner
      flex 1 1 auto
</style>
