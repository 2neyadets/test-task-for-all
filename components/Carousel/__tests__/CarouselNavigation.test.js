import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import CarouselNavigation from '../CarouselNavigation'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('CarouselNavigation component', () => {
  let wrapper
  const currentSlide = 1,
    amountOfSlides = 3

  beforeEach(() => {
    wrapper = shallowMount(CarouselNavigation, {
      localVue,
      propsData: {
        currentSlide,
        amountOfSlides,
      },
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('render correct amount of navigation btns', async () => {
    expect.assertions(2)
    expect(wrapper.findAll('MyBtn-stub').length).toBe(amountOfSlides)
    wrapper.setProps({
      amountOfSlides: 10
    })
    await localVue.nextTick()
    expect(wrapper.findAll('MyBtn-stub').length).toBe(10)
  })

  it('active slide btn have "active" class', async () => {
    expect.assertions(2)
    const navBtns = wrapper.findAll('MyBtn-stub')
    expect(navBtns.at(1).attributes('active')).toBe('true')
    wrapper.setProps({
      currentSlide: 0
    })
    await localVue.nextTick()
    expect(navBtns.at(1).attributes('active')).toBe(undefined)
  })

  it('can emit "changeSlide" event', () => {
    expect.assertions(3)
    const navBtns = wrapper.findAll('MyBtn-stub')
    navBtns.at(2).trigger('click')
    expect(wrapper.emitted('changeSlide')[0]).toEqual([2])
    navBtns.at(1).trigger('click')
    expect(wrapper.emitted('changeSlide')[1]).toEqual([1])
    navBtns.at(0).trigger('click')
    expect(wrapper.emitted('changeSlide')[2]).toEqual([0])
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
