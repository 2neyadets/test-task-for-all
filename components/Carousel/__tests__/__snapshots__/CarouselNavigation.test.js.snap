// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselNavigation component match snapshot 1`] = `
<div
  class="absolute flex carousel__navigation"
>
  <div
    class="carousel__navigation-inner flex justify-center"
  >
    <mybtn-stub
      class="mx-xs"
      icon="brightness_1"
      label=""
      round="true"
      size="12"
      type="button"
    />
    <mybtn-stub
      active="true"
      class="mx-xs"
      icon="brightness_1"
      label=""
      round="true"
      size="12"
      type="button"
    />
    <mybtn-stub
      class="mx-xs"
      icon="brightness_1"
      label=""
      round="true"
      size="12"
      type="button"
    />
  </div>
</div>
`;
